### DISCUSSION TITLE
mvn clean compile test xray:import-results - code 400

### DISCUSSION THREAD

**Discussion by fredericMagnenat** [Xray-App/xray-maven-plugin#86] [Q&A]
Hey,

I am using the xray-maven-plugin version 0.8.0 to import JUnit test results into Jira Xray. However, I have noticed that the plugin is calling an outdated API endpoint.

Currently, the plugin uses rest/raven/2.0/import/execution/junit?projectKey=XXX, but Xray's documentation recommends using the newer endpoint: rest/raven/2.0/api/import

The plugin doesn’t seem to provide a way to configure the endpoint to use this new API path. I would like to know how to resolve this issue or if there’s a way to customize the API URL in the plugin’s configuration.

``` <plugin>
     <groupId>app.getxray</groupId>
        <artifactId>xray-maven-plugin</artifactId>
          <version>0.8.0</version>
          <configuration>
            <cloud>false</cloud>
            <jiraBaseUrl>xxxxxxx</jiraBaseUrl>
            <jiraToken>xxxxxx</jiraToken>
            <projectKey>KEY01</projectKey>
            <reportFormat>junit</reportFormat>
            <reportFile>${project.basedir}/report/junit-test-repot.xml</reportFile>
         </configuration>
     </plugin>
```

Any help would be appreciated.

**Comment by bitcoder** [comment]
Hi @fredericMagnenat ,
The correct endpoint is  indeed`"rest/raven/2.0/import/execution/junit`
as detailed on the [docs](https://docs.getxray.app/display/XRAY/Import+Execution+Results+-+REST#ImportExecutionResultsREST-JUnitXMLresults); not sure where you saw that other reference; if you can share it, it would be great.

1. Can you try to submit the results using the curl utility for example? The ebst would be to get `curl` (or similar) working to have a working example and then compare it with this plugin
 
Something like 
```bash
curl -H "Content-Type: multipart/form-data" -u admin:admin -F "file=@report.xml" "http://yourserver/rest/raven/2.0/import/execution/junit?projectKey=XTP"
```

3. Is your project key "KEY01" ?
4. Can you enable debug using verbose mode and share the output?

```bash
mvn clean compile test xray:import-results  -Dxray.verbose=true -X
```

**Reply by bitcoder** [reply]
@fredericMagnenat was your issue solved? did you try the previous steps?

**Reply by fredericMagnenat** [reply]
Hey @bitcoder,
I apologize for my late reply. I was on holidays.

On my project, we use JIRA version 9.12.2 and Xray version 7.8.2-j9. 
On the Xray website, the Raven V2 API is located at "...rest/raven/2.0/api".
https://docs.getxray.app/display/XRAY/v2.0#/Import/post-import-execution

When I executed this cmd : mvn  xray:import-results  -Dxray.verbose=true -X

[DEBUG] Resolving plugin prefix xray from [org.apache.maven.plugins, org.codehaus.mojo]
[DEBUG] Resolved plugin prefix xray to app.getxray:xray-maven-plugin from POM ch.vs.data:data-hub-test-acceptance:jar:1.0.0
[DEBUG] Lifecycle clean -> [pre-clean, clean, post-clean]
[DEBUG] Lifecycle default -> [validate, initialize, generate-sources, process-sources, generate-resources, process-resources, compile, process-classes, generate-test-sources, process-test-sources, generate-test-resources, process-test-resources, test-compile, process-test-classes, test, prepare-package, package, pre-integration-test, integration-test, post-integration-test, verify, install, deploy]
[DEBUG] Lifecycle site -> [pre-site, site, post-site, site-deploy]
[DEBUG] === PROJECT BUILD PLAN ================================================
[DEBUG] Project:       ch.vs.data:data-hub-test-acceptance:1.0.0
[DEBUG] Dependencies (collect): []
[DEBUG] Dependencies (resolve): []
[DEBUG] Repositories (dependencies): [central (https://repo.maven.apache.org/maven2, default, releases)]
[DEBUG] Repositories (plugins)     : [central (https://repo.maven.apache.org/maven2, default, releases)]
[DEBUG] -----------------------------------------------------------------------
[DEBUG] Goal:          app.getxray:xray-maven-plugin:0.8.0:import-results (default-cli)
[DEBUG] Style:         Regular
[DEBUG] Configuration: <?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <abortOnError default-value="true">${xray.abortOnError}</abortOnError>
  <clientId>${xray.clientId}</clientId>
  <clientSecret>${xray.clientSecret}</clientSecret>
  <cloud default-value="false">false</cloud>
  <cloudApiBaseUrl default-value="https://xray.cloud.getxray.app/api/v2">${xray.cloudApiBaseUrl}</cloudApiBaseUrl>
  <ignoreSslErrors default-value="false">${xray.ignoreSslErrors}</ignoreSslErrors>
  <jiraBaseUrl>https://jira.datahub.ch</jiraBaseUrl>
  <jiraPassword>${xray.jiraPassword}</jiraPassword>
  <jiraToken>00U0MDQ3NzI0NzY1OjZul1m80000000000000</jiraToken>
  <jiraUsername>${xray.jiraUsername}</jiraUsername>
  <project default-value="${project}"/>
  <projectKey>WACO</projectKey>
  <reportFile>C:\Dev\Workspace\app\clients\data\data.test.acceptance\data-hub-test-acceptance/report/data-hub-test-acceptance.xml</reportFile>
  <reportFormat>junit</reportFormat>
  <revision>${xray.revision}</revision>
  <scope>${scope}</scope>
  <testEnvironment>${xray.testEnvironment}</testEnvironment>
  <testExecInfoJson>${xray.testExecInfoJson}</testExecInfoJson>
  <testExecKey>${xray.testExecKey}</testExecKey>
  <testInfoJson>${xray.testInfoJson}</testInfoJson>
  <testPlanKey>${xray.testPlanKey}</testPlanKey>
  <timeout default-value="50">${xray.timeout}</timeout>
  <useInternalTestProxy default-value="false">${xray.useInternalTestProxy}</useInternalTestProxy>
  <verbose default-value="false">${xray.verbose}</verbose>
  <version>${xray.version}</version>
</configuration>
[DEBUG] =======================================================================
[INFO]
[INFO] --- xray:0.8.0:import-results (default-cli) @ data-hub-test-acceptance ---
[DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=279400, ConflictMarker.markTime=36500, ConflictMarker.nodeCount=12, ConflictIdSorter.graphTime=343499, ConflictIdSorter.topsortTime=192501, ConflictIdSorter.conflictIdCount=8, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=1952699, ConflictResolver.conflictItemCount=10, DfDependencyCollector.collectTime=118782700, DfDependencyCollector.transformTime=4112500}
[DEBUG] app.getxray:xray-maven-plugin:jar:0.8.0
[DEBUG]    com.squareup.okhttp3:okhttp:jar:4.9.3:compile
[DEBUG]       com.squareup.okio:okio:jar:2.8.0:compile
[DEBUG]          org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.4.0:compile
[DEBUG]       org.jetbrains.kotlin:kotlin-stdlib:jar:1.4.10:compile
[DEBUG]          org.jetbrains:annotations:jar:13.0:compile
[DEBUG]    org.json:json:jar:20231013:compile
[DEBUG]    org.codehaus.plexus:plexus-utils:jar:3.4.1:compile
[DEBUG] Created new class realm plugin>app.getxray:xray-maven-plugin:0.8.0
[DEBUG] Importing foreign packages into class realm plugin>app.getxray:xray-maven-plugin:0.8.0
[DEBUG]   Imported:  < maven.api
[DEBUG] Populating class realm plugin>app.getxray:xray-maven-plugin:0.8.0
[DEBUG]   Included: app.getxray:xray-maven-plugin:jar:0.8.0
[DEBUG]   Included: com.squareup.okhttp3:okhttp:jar:4.9.3
[DEBUG]   Included: com.squareup.okio:okio:jar:2.8.0
[DEBUG]   Included: org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.4.0
[DEBUG]   Included: org.jetbrains.kotlin:kotlin-stdlib:jar:1.4.10
[DEBUG]   Included: org.jetbrains:annotations:jar:13.0
[DEBUG]   Included: org.json:json:jar:20231013
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:3.4.1
[DEBUG] Loading mojo app.getxray:xray-maven-plugin:0.8.0:import-results from plugin realm ClassRealm[plugin>app.getxray:xray-maven-plugin:0.8.0, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@73d16e93]
[DEBUG] Configuring mojo execution 'app.getxray:xray-maven-plugin:0.8.0:import-results:default-cli' with basic configurator -->
[DEBUG]   (f) abortOnError = true
[DEBUG]   (f) cloud = false
[DEBUG]   (f) cloudApiBaseUrl = https://xray.cloud.getxray.app/api/v2
[DEBUG]   (f) ignoreSslErrors = false
[DEBUG]   (f) jiraBaseUrl = https://jira.datahub.ch
[DEBUG]   (f) jiraToken = MjU0MDQ3NzI0NzY1OjZul1m8HmdAsUHHI29kI2KDR9sB
[DEBUG]   (f) project = MavenProject: ch.vs.data:data-hub-test-acceptance:1.0.0 @ C:\Dev\Workspace\app\clients\data\data.test.acceptance\data-hub-test-acceptance\pom.xml
[DEBUG]   (f) projectKey = WACO
[DEBUG]   (f) reportFile = C:\Dev\Workspace\app\clients\data\data.test.acceptance\data-hub-test-acceptance/report/data-hub-test-acceptance.xml
[DEBUG]   (f) reportFormat = junit
[DEBUG]   (f) timeout = 50
[DEBUG]   (f) useInternalTestProxy = false
[DEBUG]   (f) verbose = true
[DEBUG] -- end configuration --
[DEBUG] cloud from config: false
[DEBUG] jiraBaseUrl from config: https://jira.datahub.ch
[DEBUG] testInfoJson from config: null
[DEBUG] useInternalTestProxy from config: false
[INFO] reportFile: C:\Dev\Workspace\app\clients\data\data.test.acceptance\data-hub-test-acceptance/report/data-hub-test-acceptance.xml
[DEBUG] REQUEST_URL: https://jira.datahub.ch/rest/raven/2.0/import/execution/junit?projectKey=WACO
[DEBUG] REQUEST_METHOD: POST
[DEBUG] REQUEST_CONTENT_TYPE: multipart/form-data; boundary=c8978f86-3456-49bf-a694-6c631f3006f2
[DEBUG] RESPONSE_CONTENT_TYPE:application/json;charset=UTF-8
[DEBUG] RESPONSE_HTTP_STATUS: 400
[DEBUG] RESPONSE_BODY:
[DEBUG] =======================
[DEBUG] {"error":"WACO-5 n\u0027est pas un ticket d\u0027exigence ou n\u0027est pas une clé de ticket valide."}
[ERROR]
java.io.IOException: Unexpected HTTP code Response{protocol=http/1.1, code=400, message=, url=https://jira.datahub.ch/rest/raven/2.0/import/execution/junit?projectKey=WACO}
    at app.getxray.xray.XrayResultsImporter.makeHttpRequest (XrayResultsImporter.java:350)
    at app.getxray.xray.XrayResultsImporter.submitStandardServerDC (XrayResultsImporter.java:443)
    at app.getxray.xray.XrayResultsImporter.submit (XrayResultsImporter.java:293)
    at app.getxray.maven.plugin.xray.ImportResultsMojo.execute (ImportResultsMojo.java:254)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:569)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
[ERROR] Unexpected HTTP code Response{protocol=http/1.1, code=400, message=, url=https://jira.datahub.ch/rest/raven/2.0/import/execution/junit?projectKey=WACO}




Thank you for your help !!!

**Comment by bitcoder** [comment]
As per your logs, it seems you are referring to a issue in Jira that is not valid: WACO-5. Can you check that? Is it a Test issue or a Test Execution issue? It would also help out if you could share the JUnit XML file.

" {"error":"WACO-5 n\u0027est pas un ticket d\u0027exigence ou n\u0027est pas une clé de ticket valide."}"´

Anyway, it doesnt seem to be an issue of this plugin but more a usage related thing.

**Reply by fredericMagnenat** [reply]
@bitcoder 
I'm so sorry !!!!!!!  :-(

Thank you for everything.

**Reply by fredericMagnenat** [reply]
I think the problem must be in our Jira configuration.

**Reply by bitcoder** [reply]
Ok, I'll close this then in few days if nothing else pops up
